src/offset.js

```diff
@@ -57,10 +57,10 @@ jQuery.offset = {
 		}
 
 		if ( options.top != null ) {
-			props.top = ( options.top - curOffset.top ) + curTop;
+			props.top = ( options.top - ( curOffset ? curOffset.top : 0 ) ) + curTop;
 		}
 		if ( options.left != null ) {
-			props.left = ( options.left - curOffset.left ) + curLeft;
+			props.left = ( options.left - ( curOffset ? curOffset.left : 0 ) ) + curLeft;
```

Comment:
Invocation of ```getBoundingClientRect()``` does not mean that layout will be called each time when it's invoked, only if there are uncalculated changes (also I believe there is not real layout for truly hidden or disconnected elements).
It's okay and I can change that code if you prefer that. But it's strange to hear about that one slow operation (relatively slow to another which is faster) does not matter because there around are other slow operations.
Also I saw application of dozens of thousands simple calculations, where was no such things like layout, but they were incredibly slow just because every operation matters.

Okay, that is just thoughts and I will change that part of code later.